src/random.cpp

```diff
@@ -12,6 +12,7 @@
 #include <wincrypt.h>
 #endif
 #include <logging.h>  // for LogPrint()
+#include <utilstrencodings.h> // for ARRAYLEN
```

Comment:
nit: This is only needed when within the body of the `#elif defined(HAVE_SYSCTL_ARND)` guard, so you might as well guard the include.

Reply:
There's an existing block at line 36.